zrml/swaps/src/lib.rs

```diff
@@ -1475,11 +1471,32 @@ mod pallet {
             T::PalletId::get().into_sub_account_truncating((*pool_id).saturated_into::<u128>())
         }
 
-        // The minimum allowed balance in a liquidity pool.
+        /// The minimum allowed balance of `asset` in a liquidity pool.
         pub(crate) fn min_balance(asset: Asset<MarketIdOf<T>>) -> BalanceOf<T> {
             T::AssetManager::minimum_balance(asset).max(MIN_BALANCE.saturated_into())
         }
 
+        /// Returns the minimum allowed balance allowed for a pool with id `pool_id` containing
+        /// `assets`.
+        ///
+        /// The minimum allowed balance is the maximum of all minimum allowed balances of assets
+        /// contained in the pool, _including_ the pool shares asset. This ensures that none of the
+        /// accounts involved are slashed when a pool is created with the minimum amount.
+        ///
+        /// **Should** only be called if `assets` is non-empty. Note that the existence of a pool
+        /// with the specified `pool_id` is not mandatory.
+        pub(crate) fn min_balance_of_pool(
+            pool_id: PoolId,
+            assets: &[Asset<MarketIdOf<T>>],
+        ) -> BalanceOf<T> {
+            assets
+                .iter()
+                .map(|asset| Self::min_balance(*asset))
+                .max()
+                .unwrap_or_else(|| MIN_BALANCE.saturated_into())
+                .max(Self::min_balance(Self::pool_shares_id(pool_id)))
+        }
```

Comment:
Do you know where the existential deposits of the pool assets are set (`min_balance`)? If yes, can you please share me the reference?

Reply:
`minimum_balance` is just the `ExistentialDeposit`, which is defined in `runtime/*/parameters.rs`.

Reply:
```rust
    // Existential deposits used by orml-tokens.
    // Only native ZTG and foreign assets should have an existential deposit.
    // Winning outcome tokens are redeemed completely by the user, losing outcome tokens
    // are cleaned up automatically. In case of scalar outcomes, the market account can have dust.
    // Unless LPs use `pool_exit_with_exact_asset_amount`, there can be some dust pool shares remaining.
    // Explicit match arms are used to ensure new asset types are respected.
    pub ExistentialDeposits: |currency_id: CurrencyId| -> Balance {
        match currency_id {
            Asset::CategoricalOutcome(_,_) => ExistentialDeposit::get(),
            Asset::CombinatorialOutcome => ExistentialDeposit::get(),
            Asset::PoolShare(_)  => ExistentialDeposit::get(),
            Asset::ScalarOutcome(_,_)  => ExistentialDeposit::get(),
            #[cfg(feature = "parachain")]
            Asset::ForeignAsset(id) => {
                let maybe_metadata = <
                    orml_asset_registry::Pallet<super::Runtime> as orml_traits::asset_registry::Inspect
                >::metadata(&Asset::ForeignAsset(*id));

                if let Some(metadata) = maybe_metadata {
                    return metadata.existential_deposit;
                }

                1
            }
            #[cfg(not(feature = "parachain"))]
            Asset::ForeignAsset(_) => ExistentialDeposit::get(),
            Asset::Ztg => ExistentialDeposit::get(),
        }
    };
```

Ah fine, got it thanks!